static/app/views/performance/transactionSummary/transactionTags/tagValueTable.tsx

```diff
@@ -0,0 +1,319 @@
+
+    const currentSortKind = currentSort ? currentSort.kind : undefined;
+
+    return (
+      <SortLink
+        align="left"
+        title={columnInfo.name}
+        direction={currentSortKind}
+        canSort={canSort}
+        generateSortLink={generateSortLink}
+        onClick={() => {}} // TODO(k-fish): Implement sorting
+      />
+    );
+  }
+
+  renderHeadCellWithMeta = (
+    sortedEventView: EventView,
+    tableMeta: TableData['meta'],
+    columns: TagColumn[]
+  ) => {
+    return (column: TableColumn<ColumnKeys>, index: number): React.ReactNode =>
+      this.renderHeadCell(sortedEventView, tableMeta, column, columns[index]);
+  };
+
+  handleTagValueClick = (location: Location, tagKey: string, tagValue: string) => {
+    const queryString = decodeScalar(location.query.query);
+    const conditions = tokenizeSearch(queryString || '');
+
+    conditions.addTagValues(tagKey, [tagValue]);
+
+    const query = stringifyQueryObject(conditions);
+    browserHistory.push({
+      pathname: location.pathname,
+      query: {
+        ...location.query,
+        query: String(query).trim(),
+      },
+    });
+  };
+
+  handleCellAction = (
+    column: TableColumn<ColumnKeys>,
+    tagValue: React.ReactText,
+    actionRow: any
+  ) => {
+    return (action: Actions) => {
+      const {eventView, location} = this.props;
+
+      const searchConditions = tokenizeSearch(eventView.query);
+
+      searchConditions.removeTag('event.type');
+
+      updateQuery(searchConditions, action, {...column, name: actionRow.id}, tagValue);
+
+      browserHistory.push({
+        pathname: location.pathname,
+        query: {
+          ...location.query,
+          [TAGS_CURSOR_NAME]: undefined,
+          query: stringifyQueryObject(searchConditions),
+        },
+      });
+    };
+  };
+
+  renderBodyCell = (
+    parentProps: Props,
+    column: TableColumn<ColumnKeys>,
+    dataRow: TableDataRow
+  ): React.ReactNode => {
+    const value = dataRow[column.key];
+    const {location} = parentProps;
+
+    if (column.key === 'key') {
+      return dataRow.tags_key;
+    }
+
+    const allowActions = [Actions.ADD, Actions.EXCLUDE];
+
+    if (column.key === 'tagValue') {
+      const actionRow = {...dataRow, id: dataRow.tags_key};
+      return (
+        <CellAction
+          column={column}
+          dataRow={actionRow}
+          handleCellAction={this.handleCellAction(column, dataRow.tags_value, actionRow)}
+          allowActions={allowActions}
+        >
+          <TagValue row={dataRow} />
+        </CellAction>
+      );
+    }
+
+    if (column.key === 'frequency') {
+      return formatPercentage(dataRow.frequency, 0);
+    }
+
+    if (column.key === 'action') {
+      return (
+        <Link
+          to=""
+          onClick={() =>
+            this.handleTagValueClick(location, dataRow.tags_key, dataRow.tags_value)
+          }
+        >
+          <LinkContainer>
+            <IconAdd isCircled /> {t('Add to filter')}
```

Comment:
Feedback: This link/action should be disabled once clicked. Adding additional duplicated search conditions won't be helpful.

Reply:
That's a good call.